src/solver/optimisation/AreaBalance.cpp

```diff
@@ -0,0 +1,74 @@
+#include "AreaBalance.h"
+
+static void shortTermStorageBalance(
+  const ::ShortTermStorage::AREA_INPUT& shortTermStorageInput,
+  ConstraintBuilder& constraintBuilder,
+  const CORRESPONDANCES_DES_VARIABLES& CorrespondanceVarNativesVarOptim)
+{
+    for (const auto& storage : shortTermStorageInput)
+    {
+        unsigned index = storage.clusterGlobalIndex;
+        if (const int varInjection
+            = CorrespondanceVarNativesVarOptim.SIM_ShortTermStorage.InjectionVariable[index];
```

Comment:
I think this is redundant with the check done inside the builder.

Then if we remove it, we should not need anymore the `CORRESPONDANCES_DES_VARIABLES`.